packages/web/pages/technology/[id]/edit.js

```diff
@@ -203,26 +209,30 @@ const TechnologyFormPage = ({ taxonomies, technology }) => {
 
 	return (
 		<ContentContainer bgColor={colors.gray98}>
-			<Protected>
-				<Title align="left" noPadding noMargin>
-					Cadastrar <span>Tecnologia</span>
-				</Title>
-
-				<FormWizard
-					// forces a re-render to populate defaultValues, otherwise we would need to call reset()
-					key={currentStep}
-					onSubmit={handleSubmit}
-					onPrev={handlePrev}
-					currentStep={currentStep || techonologyFormSteps[0].slug}
-					submitting={submitting}
-					steps={techonologyFormSteps}
-					data={{
-						taxonomies,
-						technology,
-					}}
-					defaultValues={technology}
-				/>
-			</Protected>
+			{authorizedToEdit ? (
+				<Protected>
+					<Title align="left" noPadding noMargin>
+						Cadastrar <span>Tecnologia</span>
+					</Title>
+
+					<FormWizard
+						// forces a re-render to populate defaultValues, otherwise we would need to call reset()
+						key={currentStep}
+						onSubmit={handleSubmit}
+						onPrev={handlePrev}
+						currentStep={currentStep || techonologyFormSteps[0].slug}
+						submitting={submitting}
+						steps={techonologyFormSteps}
+						data={{
+							taxonomies,
+							technology,
+						}}
+						defaultValues={technology}
+					/>
+				</Protected>
+			) : (
+				<NotAuthorized messageContext="techStatus" />
```

Comment:
This doesn't seem like a good use case for the `Protected` component. It was developed precisely to handle the authorization to show the content **or** an unauthorized message.

I think it would be better to modify the code for the `Protected` component to receive a customized validation (the boolean `authorizedToEdit`) and, based on this validation, it decides whether to display the `NotAuthorized` component or not.

For me, the best use case for this component would be something like:
```jsx
<Protected customRule={authorizedToEdit} unauthorizedMessage="techStatus">
  {/** content to be hidden based on the viewing rule */}
</Protected>
```

Reply:
Makes sense